src/caret_analyze/plot/visualize_lib/bokeh/bokeh.py

```diff
@@ -44,6 +47,239 @@ class Bokeh(VisualizeLibInterface):
+            raise NotImplementedError()
+
+        color_selector = ColorSelectorFactory.create_instance(coloring_rule='unique')
+        legend_manager = LegendManager()
+        stacked_bar_source = StackedBarSource(legend_manager, target_objects, 0.0, xaxis_type)
+        fig.add_tools(stacked_bar_source.create_hover())
+        stacked_bar_data, x_width_list = \
+            self._get_stacked_bar_data(data, y_labels, xaxis_type, x_label)
+        bottom_labels = self._get_bottom_labels(y_labels)
+        bottom_labels = bottom_labels[1:]
+        source = stacked_bar_source.generate(
+            target_objects,
+            stacked_bar_data,
+            y_labels,
+            bottom_labels,
+            x_width_list
+        )
+
+        for y_label, bottom in zip(y_labels[:-1], bottom_labels):
+            color = color_selector.get_color(y_label)
+            renderer = fig.vbar(
+                x=x_label,
+                top=y_label,
+                width='x_width_list',
+                source=source,
+                color=color,
+                bottom=bottom,
+                # legend_label=y_label,
+            )
+            legend_manager.add_legend(y_label, renderer, y_label)
+            # color = next(color_generator)
+        color = color_selector.get_color(y_labels[-1])
+        renderer = fig.vbar(
+            x=x_label,
+            top=y_labels[-1],
+            width='x_width_list',
+            source=source,
+            color=color,
+            # legend_label=y_labels[-1],
+        )
+        legend_manager.add_legend(y_labels[-1], renderer, y_labels[-1])
+
+        num_legend_threshold = 20
+        legends = legend_manager.create_legends(num_legend_threshold, full_legends)
+        for legend in legends:
+            fig.add_layout(legend, 'right')
+        fig.legend.click_policy = 'mute'
+
+        return fig
+
+    def _get_stacked_bar_data(
+        self,
+        data: Dict[str, List[int]],
+        y_labels: List[str],
+        xaxis_type: str,
+        x_label: str,
+    ) -> Tuple[Dict[str, List[float]],  List[float]]:
+        """
+        Calculate stacked bar data.
+
+        Parameters
+        ----------
+        data : Dict[str, List[int]]
+            Source data.
+        y_labels : List[str]
+            Y axis labels that are Node/Topic name.
+        xaxis_type : str
+            Type of x-axis of the line graph to be plotted.
+            "system_time", "index", or "sim_time" can be specified.
+            The default is "system_time".
+        x_label : str
+            X-axis label of data dict.
+            "start time" or "index".
+
+        Returns
+        -------
+        Dict[str, List[float]]
+            Stacked bar data.
+        List[float]
+            Width list of bars.
+
+        """
+        output_data: Dict[str, List[float]] = {}
+        x_width_list: List[float] = []
+
+        # Convert the data unit to second
+        output_data = self._update_data_unit(data, 1e-9)
```

Comment:
Multiple functions, such as `_update_data_unit`, are called like `member function` despite `static method`.

This is undesirable because it appears that `self` has been changed.
As an example, Line 179 should be changed as follows:
```suggestion
        # Convert the data unit to second
        output_data = Bokeh._update_data_unit(data, 1e-9)
```

Also, `_update_data_unit` should be renamed.
See comment below for details.

Reply:
I changed all staticmethod to Bokeh.